docs/application/native/guides/text-input/voice-control-engine.md

```diff
@@ -0,0 +1,527 @@
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+- Get current audio type using the `vce_get_audio_type()` function.
+
+  ```
+    void
+    get_audio_type(void)
+    {
+        int ret = VCE_ERROR_NONE;
+        char *audio_type = NULL;
+
+        ret = vce_get_audio_type(&audio_type);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+        if(NULL != audio_type) {
+            /* Memory release */
+            free(audio_type);
+            audio_type = NULL;
+        }
+    }
+  ```
+
+- Get private data from a voice manager client using the `vce_get_private_data()` function.
+
+  ```
+    void
+    get_private_data(void)
+    {
+        int ret = VCE_ERROR_NONE;
+        char *data = NULL;
+
+        ret = vce_get_private_data("key", &data);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+        if(NULL != data) {
+            /* Memory release */
+            free(data);
+            data = NULL;
+        }
+    }
+  ```
+
+
+<a name="control_record"></a>
+## VCE Control Recording
+
+You can control recording of the following information about the VCE:
+
+- Start recording voice using the `vce_start_recording()` function. If the function call is successful, You can receive recording data from `set_recording` function.
+
+  ```
+    void
+    start_recording(void)
+    {
+        int ret = VCE_ERROR_NONE;
+
+        ret = vce_start_recording();
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+- Stop recording voice using the `vce_stop_recording()` function. If the function call is successful, You can not receive recording data from `set_recording` function.
+
+  ```
+    void
+    stop_recording(void)
+    {
+        int ret = VCE_ERROR_NONE;
+
+        ret = vce_stop_recording();
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+<a name="send_result"></a>
+## VCE Send Results
+
+You can send the following result information about the VCE:
+
+- Send the command result to the engine service user using the `vce_send_result()` function.
+
+  ```
+    void
+    send_result(void)
+    {
+        int ret = VCE_ERROR_NONE;
+        int is_consumed;
+
+        ret = vce_send_result(VCE_RESULT_EVENT_SUCCESS, NULL, 0, "text", NULL, NULL, VC_RESULT_MESSAGE_NONE, &is_consumed, NULL);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+- Send the Automatic Speech Recognition (ASR) result to the engine service user using the vce_send_asr_result() function.
+
+  ```
+    void
+    send_asr_result(void)
+    {
+        int ret = VCE_ERROR_NONE;
+        ret = vce_send_asr_result(VCE_ASR_RESULT_EVENT_FINAL_RESULT, "asr_result", NULL);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+- Send the Natural Language Generation (NLG) result to the engine service user using the vce_send_nlg_result() function.
```

Comment:
Send the Natural Language Generation (NLG) result to the engine service user using the vce_send_nlg_result() function.
Please change the sentence as:
Send the Natural Language Generation (NLG) result to the engine service user using the `vce_send_nlg_result()` function.

Reply:
Done.